executor.go

```diff
@@ -223,15 +224,30 @@ func (e *Executor) Wait(ctx context.Context) (*ExecResult, error) {
 	}
 
 	if e.capture {
-		res.Stdout = string(e.stdoutBuf.Bytes())
-		res.Stderr = string(e.stderrBuf.Bytes())
+		res.Stdout = e.stdoutBuf.String()
```

Comment:
with this, I don't quite understand what the pipe writing is for below if we do this. e.stdoutBuf will not contain the flushed output, e.stdout will.